armory/data/datasets.py

```diff
@@ -88,7 +101,7 @@ def __next__(self):
         return self.get_batch()
 
     def __len__(self):
-        return self.size
+        return self.batches_per_epoch * self.epochs
```

Comment:
For tqdm we want to show a total number of batches. Previously it was the size (i.e. number of samples per epoch) which doesn't match the iteration